File: algorithm/programmers_영어끝말잇기.js
function solution(n, words) {
    const answer = [0, 0];
    const answerSet = new Set();
    answerSet.add(words[0]);
    for (let i = 1; i < words.length; i++) {
        const prev = words[i - 1];
        const cur = words[i];        
        if (answerSet.has(cur) || prev[prev.length - 1] !== cur[0]) {
            return [i % n + 1, Math.ceil((i+1) / n)];
        }
        answerSet.add(cur);
    }

    return answer;
}

File: algorithm/programmers_영어끝말잇기.py
import math

def solution(n, words):
    answerSet = set([words[0]])
    
    for idx in range(len(words)):
        if idx == 0:
            continue
        prev = words[idx - 1]
        cur = words[idx]
        if prev[len(prev) - 1] != cur[0] or cur in answerSet:
            return [idx % n + 1, math.ceil((idx + 1) / n)]
        answerSet.add(cur)
        
    return [0, 0]
